src/classes/aflightentry.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ *openPilot Log - A FOSS Pilot Logbook Application
+ *Copyright (C) 2020  Felix Turowsky
+ *
+ *This program is free software: you can redistribute it and/or modify
+ *it under the terms of the GNU General Public License as published by
+ *the Free Software Foundation, either version 3 of the License, or
+ *(at your option) any later version.
+ *
+ *This program is distributed in the hope that it will be useful,
+ *but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *GNU General Public License for more details.
+ *
+ *You should have received a copy of the GNU General Public License
+ *along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "aflightentry.h"
+#include "src/database/adatabase.h"
+
+AFlightEntry::AFlightEntry()
+    : AEntry::AEntry(DEFAULT_FLIGHT_POSITION)
+{}
+
+AFlightEntry::AFlightEntry(int row_id)
+    : AEntry::AEntry(DataPosition(DB_TABLE_FLIGHTS, row_id))
+{}
+
+AFlightEntry::AFlightEntry(RowData table_data)
+    : AEntry::AEntry(DEFAULT_FLIGHT_POSITION, table_data)
+{}
+
+const QString AFlightEntry::summary()
+{
+    if(tableData.isEmpty())
+        return QString();
+
+    QString flight_summary;
+    flight_summary.append(tableData.value(DB_FLIGHTS_DOFT).toString() + " ");
+    flight_summary.append(tableData.value(DB_FLIGHTS_DEPT).toString() + " ");
+    flight_summary.append(ACalc::minutesToString(tableData.value(DB_FLIGHTS_TOFB).toString()) + " ");
+    flight_summary.append(ACalc::minutesToString(tableData.value(DB_FLIGHTS_TONB).toString()) + " ");
+    flight_summary.append(tableData.value(DB_FLIGHTS_DEST).toString() + " ");
+
+    return flight_summary;
+}
+
+const QString AFlightEntry::getRegistration()
+{
+    ATailEntry acft = aDB()->resolveForeignTail(tableData.value(DB_FLIGHTS_ACFT).toInt());
+    return acft.registration();
+}
+
+const QString AFlightEntry::getPilotName(pilot pilot_)
+{
+    switch (pilot_) {
+    case pilot::pic: {
+        auto foreign_pilot = aDB()->resolveForeignPilot(tableData.value(DB_FLIGHTS_PIC).toInt());
+        return foreign_pilot.name();
+        break;
+    }
+    case pilot::sic: {
+        auto foreign_pilot = aDB()->resolveForeignPilot(tableData.value(DB_FLIGHTS_SECONDPILOT).toInt());
+        return foreign_pilot.name();
+    }
+    case pilot::thirdPilot: {
+        auto foreign_pilot = aDB()->resolveForeignPilot(tableData.value(DB_FLIGHTS_THIRDPILOT).toInt());
+        return foreign_pilot.name();
+        break;
+    } // case scope
+    } // switch (pilot_)
+    return QString();
+}
```

Comment:
This is a bit suspect use of the enum.
`enum something {...}` is a plain enum. Even though you can write `something::A` you are misusing the scope operator. Enum classes are prefered over enums in general because they *demand* the use of the `::` operator and are more type safe. Because they `pilots` enum is already encapsulated the risk isnt too great but at least show that its plain in its use.
```c++
switch(pilot_) {
    case AFlightEntry::pic:  // explicit
    case sic: // implicit since you are working in the method scope (in this context this is ok)
```
If it was enum class...
```c++
switch(pilot_) {
    case AFlightEntry::pilots::pic:  // explicit root
    case pilots::sic:  // implicit since in method
    case sic:  // wont compile
```